haskell-tests/Types/TimestampSpec.hs

```diff
@@ -0,0 +1,46 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Basic JSON serialization and deserialization tests for 'Timestamp'.
+module Types.TimestampSpec where
+
+import Control.Monad
+import qualified Data.Aeson as AE
+import qualified Data.ByteString.Lazy as LBS
+import Test.Hspec
+import Test.QuickCheck
+
+import Concordium.Common.Time
+
+import Generators
+
+testEncodeDecode :: Property
+testEncodeDecode = forAll genTimestamp $ \ts -> AE.decode (AE.encode ts) === Just ts
+
+decodeExamples :: [(LBS.ByteString, Maybe Timestamp)]
+decodeExamples =
+    [ ("0", Just $ Timestamp 0),
+      ("1", Just $ Timestamp 1),
+      ("1.0", Just $ Timestamp 1),
```

Comment:
This would not be a valid timestamp though... right? I feel like it would be best to limit the tests to only expected positives/negatives.. or am I misunderstanding something?